collector/pg_stat_statements.go

```diff
@@ -40,41 +40,42 @@ func NewPGStatStatementsCollector(config collectorConfig) (Collector, error) {
 }
 
 var (
-	statSTatementsCallsTotal = prometheus.NewDesc(
+	statStatementsCallsTotal = prometheus.NewDesc(
 		prometheus.BuildFQName(namespace, statStatementsSubsystem, "calls_total"),
 		"Number of times executed",
-		[]string{"user", "datname", "queryid"},
+		[]string{"user", "datname", "queryid", "query"},
 		prometheus.Labels{},
 	)
 	statStatementsSecondsTotal = prometheus.NewDesc(
 		prometheus.BuildFQName(namespace, statStatementsSubsystem, "seconds_total"),
 		"Total time spent in the statement, in seconds",
-		[]string{"user", "datname", "queryid"},
+		[]string{"user", "datname", "queryid", "query"},
 		prometheus.Labels{},
 	)
 	statStatementsRowsTotal = prometheus.NewDesc(
 		prometheus.BuildFQName(namespace, statStatementsSubsystem, "rows_total"),
 		"Total number of rows retrieved or affected by the statement",
-		[]string{"user", "datname", "queryid"},
+		[]string{"user", "datname", "queryid", "query"},
 		prometheus.Labels{},
 	)
 	statStatementsBlockReadSecondsTotal = prometheus.NewDesc(
 		prometheus.BuildFQName(namespace, statStatementsSubsystem, "block_read_seconds_total"),
 		"Total time the statement spent reading blocks, in seconds",
-		[]string{"user", "datname", "queryid"},
+		[]string{"user", "datname", "queryid", "query"},
 		prometheus.Labels{},
 	)
 	statStatementsBlockWriteSecondsTotal = prometheus.NewDesc(
 		prometheus.BuildFQName(namespace, statStatementsSubsystem, "block_write_seconds_total"),
 		"Total time the statement spent writing blocks, in seconds",
-		[]string{"user", "datname", "queryid"},
+		[]string{"user", "datname", "queryid", "query"},
 		prometheus.Labels{},
 	)
 
 	pgStatStatementsQuery = `SELECT
 		pg_get_userbyid(userid) as user,
 		pg_database.datname,
 		pg_stat_statements.queryid,
+		pg_stat_statements.query as query,
```

Comment:
I would highly recommend we limit the length of this string. The stored query could be _very_ long.

We should add a flag like this:
```
--collector.stat_statements.query_text_limit=120
```

Then we can do something like this:
```suggestion
		LEFT(pg_stat_statements.query, %d) as query,
```

Reply:
@fireflg asked me to help with this. I've looked at the code, this feature could lead to significant refactoring, as there is no way to set individual options to collectors. Can we make the constant limit for now?

Reply:
@SuperQ

Reply:
@SuperQ I would also think that a hardcoded limit might meet the needs of many users already, provided that a significant refactoring would be necessary to pass the flag to the collector.